sequence_processing_pipeline/tests/test_NuQCJob.py

```diff
@@ -1101,474 +1102,131 @@ def test_completed_file_generation_some_failures(self):
-        '#!/bin/bash',
-        ('#SBATCH --job-name abcdabcdabcdabcdabcdabcdabcdabcd_QCJob_NYU_BMS'
-         '_Melanoma_13059'),
-        '#SBATCH -p queue_name',
-        '#SBATCH -N 1',
-        '#SBATCH -n 16',
-        '#SBATCH --time 1440',
-        '#SBATCH --mem 8gb',
-        '#SBATCH --array 1-250%30',
-        'set -x',
-        'date',
-        'hostname',
-        'echo ${SLURM_JOBID} ${SLURM_ARRAY_TASK_ID}',
-        'cd sequence_processing_pipeline/tests/data/output_dir/QCJob',
-        'offset=${SLURM_ARRAY_TASK_ID}',
-        'step=$(( $offset - 0 ))',
-        'cmd0=$(head -n $step sequence_processing_pipeline/tests/data/output_d'
-        'ir/QCJob/QCJob_NYU_BMS_Melanoma_13059.array-details | tail -n 1)',
-        'eval $cmd0',
-        ('echo "Cmd Completed: $cmd0" > logs/'
-         'QCJob_NYU_BMS_Melanoma_13059_$step.completed')]
+        demux_bin_rgx = re.compile(r'\s+(.*/bin)/demux \\')
 
-    exp_QCJob_2_chained = [
-        '#!/bin/bash',
-        ('#SBATCH --job-name abcdabcdabcdabcdabcdabcdabcdabcd_QCJob_'
-         'Feist_11661'),
-        '#SBATCH -p queue_name',
-        '#SBATCH -N 1',
-        '#SBATCH -n 16',
-        '#SBATCH --time 1440',
-        '#SBATCH --mem 8gb',
-        '#SBATCH --array 1-250%30',
-        'set -x',
-        'date',
-        'hostname',
-        'echo ${SLURM_JOBID} ${SLURM_ARRAY_TASK_ID}',
-        'cd sequence_processing_pipeline/tests/data/output_dir/QCJob',
-        'offset=${SLURM_ARRAY_TASK_ID}',
-        'step=$(( $offset - 0 ))',
-        'cmd0=$(head -n $step sequence_processing_pipeline/tests/data/output_d'
-        'ir/QCJob/QCJob_Feist_11661.array-details | tail -n 1)',
-        'eval $cmd0',
-        'echo "Cmd Completed: $cmd0" > logs/QCJob_Feist_11661_$step.completed']
+        for i in range(0, len(obs)):
+            m = base_path_rgx.search(obs[i])
+            if m:
+                obs[i] = obs[i].replace(m[1], 'REMOVED')
+            else:
+                m = demux_bin_rgx.search(obs[i])
+                if m:
+                    obs[i] = obs[i].replace(m[1], 'REMOVED')
 
-    exp_QCJob_3_chained = [
-        '#!/bin/bash',
-        ('#SBATCH --job-name abcdabcdabcdabcdabcdabcdabcdabcd_QCJob_'
-         'Gerwick_6123'),
-        '#SBATCH -p queue_name',
-        '#SBATCH -N 1',
-        '#SBATCH -n 16',
-        '#SBATCH --time 1440',
-        '#SBATCH --mem 8gb',
-        '#SBATCH --array 1-9%30',
-        'set -x',
-        'date',
-        'hostname',
-        'echo ${SLURM_JOBID} ${SLURM_ARRAY_TASK_ID}',
-        'cd sequence_processing_pipeline/tests/data/output_dir/QCJob',
-        'offset=${SLURM_ARRAY_TASK_ID}',
-        'step=$(( $offset - 0 ))',
-        'cmd0=$(head -n $step sequence_processing_pipeline/tests/data/output_d'
-        'ir/QCJob/QCJob_Gerwick_6123.array-details | tail -n 1)',
-        'eval $cmd0',
-        ('echo "Cmd Completed: $cmd0" > logs/'
-         'QCJob_Gerwick_6123_$step.completed')]
+        exp = ["#!/bin/bash -l",
```

Comment:
This is a complex to maintain approach

Reply:
Right now it's a exact-match comparison but moving forward, we can alter the test to spot check just the important information. We should be receiving a new version shortly that will help identify what we can expect to see change from version to version. All versions will need to generate a job script so it's a good place to test the black box, so to speak.

Reply:
We don't need to test that the rendering engine works, as that is the responsibility of jinja2 which is well tested. This doesn't need to block _this_ PR, but an alternative of testing the presence of specific pieces of text will be *significantly* easier to maintain. Remember that we need to be able to make improvements to the exact submission rapidly as we identify ways to increase throughput.

Reply:
> We don't need to test that the rendering engine works, as that is the responsibility of jinja2 which is well tested. This doesn't need to block _this_ PR, but an alternative of testing the presence of specific pieces of text will be _significantly_ easier to maintain. Remember that we need to be able to make improvements to the exact submission rapidly as we identify ways to increase throughput.

Right! I think we're all in agreement on this one.